sematic/ui/src/pipelines/RunPanelContext.tsx

```diff
@@ -0,0 +1,10 @@
+import React from "react";
+
+export const RunPanelContext = React.createContext<{
```

Comment:
Can you help explain why you choose to make a new context instead of merging to existing `usePipelinePanelsContext`?

I am worried about creating too many fragments and complexity with little performance gain.

Reply:
The pipeline panel context is creating the context required for React changes related to the side panel. Since the run tab and its input selection are outside of the pipeline panel, I want to separate the logic for it into a different object.

Reply:
Hi @kamalesh0406 I see what you mean now. Here are my 2 cents:

The `pipelinePanelsContext` are for all three panels (left, center, and right, as its plural name suggests). If it is only to serve the side(left) panel, it shouldn't be up to `<PipelinePanels>` (which parent `<MenuPanel />`(L), `<RunPanel />`(C) and `<NotesPanel />`(R)) to create it. As a matter of fact, the `<RunPanel />` and its descendants intensively use this context. E.g. the run node selection on the graph will determine the state of the run tree selection. So it is hard to say whether the concept of `runSelected` belongs to the side panel or the center panel. It is a shared concept of all panels. Each panel is a reflection of this shared state.

This is how I interpret these concepts. And I guess it is ok to merge your stuff together into this context as well.

Reply:
Right, so the state like `runSelected` is shared by multiple panels i.e. the Menu Panel and the Run Panel. But the `selectedRunTab` and the `selectedInputTabArtifact` are used only by the Run Panel, that's why I thought of using a separate context. 

I will go ahead with your suggestion and merge this into the `PipelinePanelsContext`.

Reply:
Yes, I understand. Thank you. I really appreciate that :)